packages/core/register/src/hook.js

```diff
@@ -15,21 +15,50 @@ const DEFAULT_CLI_OPTS = {
 };
 
 let hooks = {};
-let parcelDeps = [];
+let parcelModuleDirs = [];
+
+function getPkgUp(filename) {
+  return syncPromise(loadConfig(filename, ['package.json']));
+}
+
+function getModuleDir(filename) {
+  let pkg = getPkgUp(filename);
+  if (!pkg || !pkg.files[0]) {
+    return null;
+  }
+
+  return path.dirname(pkg.files[0].filePath);
+}
+
+function isParcelModule(filename) {
+  return !!parcelModuleDirs.find(moduleDir => filename.startsWith(moduleDir));
```

Comment:
This could also be return 

```Javascript
return parcelModuleDirs.includes(getModuleDir(filename));
```

But that's probably not as good for performance as it uses the fs instead of doing simple string comparisons.